services/product-service/src/types/types.ts

```diff
@@ -41,3 +41,28 @@ export interface CreateProductRequestBody {
   price: number;
   count: number;
 }
+
+export type Mock<T> = {
+  service: T;
+  functions: {
+    // @ts-ignore works as expected but shows error, re-investigate later
+    [P in keyof T]?: jest.Mock<ReturnType<T[P]>, Parameters<T[P]>>;
+  };
+};
+
+export const createMockService = <T>(
```

Comment:
these 2 generic type/function just helps writing unit tests, but you have not used them so they can be deleted.